macOS/DuckDuckGo/TabBar/View/TabBarViewController.swift

```diff
@@ -735,14 +599,13 @@ final class TabBarViewController: NSViewController, TabBarRemoteMessagePresentin
     private func selectTab(with event: NSEvent) {
         let locationInWindow = event.locationInWindow
 
-        if let point = pinnedTabsHostingView?.mouseLocationInsideBounds(locationInWindow),
-           let index = pinnedTabsView?.index(forItemAt: point) {
-
-            tabCollectionViewModel.select(at: .pinned(index))
-
-        } else if let point = collectionView.mouseLocationInsideBounds(locationInWindow),
-                  let indexPath = collectionView.indexPathForItem(at: point) {
+        if let indexPath = collectionView.indexPathForItemAtMouseLocation(locationInWindow) {
             tabCollectionViewModel.select(at: .unpinned(indexPath.item))
+            return
+        }
+
+        if let indexPath = pinnedTabsCollectionView?.indexPathForItemAtMouseLocation(locationInWindow) {
+            tabCollectionViewModel.select(at: .pinned(indexPath.item))
         }
```

Comment:
I've noticed we had a regression here. Fixed, test scenario detailed in the PR description